### ISSUE TITLE
Creating 256 emb vector from text during inferenece

### ISSUE THREAD

**Issue by yulies** [m-bain/frozen-in-time#47]
Dear Team,
How do I create txt embeddings for testing purposes? 
Where is the trained txt model? 
How do I load it? 
How do I run inference on the exising FAISS index from a textual input?

**Comment by m-bain** [comment]
`from transformers import AutoModel, AutoTokenizer`
`import torch.nn as nn`

To create text embeddings:
You can create a lightweight text model class, which takes text backbone and project layer:
`text_model = AutoModel.from_pretrained("distilerbert-base-uncased")`
`txt_proj = nn.Sequential(nn.ReLU(), nn.Linear(text_backbone.config.hidden_size, 256))`
Then load weights from frozen-in-time checkpoints (its under `model.text_model` and `model.txt_proj` in the checkpoint file)

1. Tokenize the list of text queries:
`tokenizer = AutoTokenizer.from_pretrained('distilbert-base-uncased')`
`text = self.tokenizer(['Example text query sentence'], return_tensors='pt', padding=True, truncation=True)`
2. Pass tokenizer output through the text backbone, and text_proj:
`text_embeddings = text_model(**text).last_hidden_state[:, 0, :]`
`text_embeddings = txt_proj(text_embeddings)`
You can then use these text embeddings to search the FAISS index.

There might be some syntax errors because I wrote this on my phone, if you are still struggling I can write a more detailed guide on building FAISS index and doing inference retrieval

**Comment by matchyc** [comment]
> `from transformers import AutoModel, AutoTokenizer` `import torch.nn as nn`
> 
> To create text embeddings: You can create a lightweight text model class, which takes text backbone and project layer: `text_model = AutoModel.from_pretrained("distilerbert-base-uncased")` `txt_proj = nn.Sequential(nn.ReLU(), nn.Linear(text_backbone.config.hidden_size, 256))` Then load weights from frozen-in-time checkpoints (its under `model.text_model` and `model.txt_proj` in the checkpoint file)
> 
> 1. Tokenize the list of text queries:
>    `tokenizer = AutoTokenizer.from_pretrained('distilbert-base-uncased')`
>    `text = self.tokenizer(['Example text query sentence'], return_tensors='pt', padding=True, truncation=True)`
> 2. Pass tokenizer output through the text backbone, and text_proj:
>    `text_embeddings = text_model(**text).last_hidden_state[:, 0, :]`
>    `text_embeddings = txt_proj(text_embeddings)`
>    You can then use these text embeddings to search the FAISS index.
> 
> There might be some syntax errors because I wrote this on my phone, if you are still struggling I can write a more detailed guide on building FAISS index and doing inference retrieval

does it mean that text embeddings can be generated from any pre-trained language models?
